tensorflow/python/framework/tensor_util_test.py

```diff
@@ -662,6 +663,69 @@ def testShapeEquals(self):
     self.assertFalse(tensor_util.ShapeEquals(t, [1, 4]))
     self.assertFalse(tensor_util.ShapeEquals(t, [4]))
 
+  def testMockArray(self):
+    class MockArray(object):
+      def __init__(self, array):
+        self.array = array
+
+      def __array__(self, dtype=None):
+        return np.asarray(self.array, dtype)
+
+    ma = MockArray(np.array([10, 20, 30]))
+    t = tensor_util.make_tensor_proto(ma)
+    if sys.byteorder == "big":  
+      self.assertProtoEquals("""  
+        dtype: DT_INT64  
+        tensor_shape { dim { size: 3 } }  
+        tensor_content: "\000\000\000\000\000\000\000\\n\000\000\000\000\000\000\000\024\000\000\000\000\000\000\000\036"  
+        """, t)  
+    else:  
+      self.assertProtoEquals("""
+        dtype: DT_INT64
+        tensor_shape { dim { size: 3 } }
+        tensor_content: "\\n\000\000\000\000\000\000\000\024\000\000\000\000\000\000\000\036\000\000\000\000\000\000\000"
+        """, t)
+    a = tensor_util.MakeNdarray(t)
+    self.assertEquals(np.int64, a.dtype)
+    self.assertAllClose(np.array([10, 20, 30], dtype=np.int64), a)
+
+  def testSeries(self):
+    df = pd.Series([10, 20, 30])
+    t = tensor_util.make_tensor_proto(df)
+    if sys.byteorder == "big":  
+      self.assertProtoEquals("""  
+        dtype: DT_INT64  
+        tensor_shape { dim { size: 3 } }  
+        tensor_content: "\000\000\000\000\000\000\000\\n\000\000\000\000\000\000\000\024\000\000\000\000\000\000\000\036"  
+        """, t)  
+    else:  
+      self.assertProtoEquals("""
+        dtype: DT_INT64
+        tensor_shape { dim { size: 3 } }
+        tensor_content: "\\n\000\000\000\000\000\000\000\024\000\000\000\000\000\000\000\036\000\000\000\000\000\000\000"
+        """, t)
+    a = tensor_util.MakeNdarray(t)
+    self.assertEquals(np.int64, a.dtype)
+    self.assertAllClose(np.array([10, 20, 30], dtype=np.int64), a)
+
+  def testDataFrame(self):
+    df = pd.DataFrame({'A': [10], 'B': [20], 'C': [30]})
```

Comment:
Let's test only a generic object with the __array__ method and comment that this supports pandas.